src/test/java/com/pmarko09/medical_clinic_proxy/service/DoctorServiceTest.java

```diff
@@ -0,0 +1,84 @@
+package com.pmarko09.medical_clinic_proxy.service;
+
+import com.pmarko09.medical_clinic_proxy.client.MedicalClinicProxyClient;
+import com.pmarko09.medical_clinic_proxy.model.dto.VisitDto;
+import feign.FeignException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+public class DoctorServiceTest {
+
+    MedicalClinicProxyClient medicalClinicProxyClient;
+    DoctorService doctorService;
+
+    @BeforeEach
+    void setup() {
+        this.medicalClinicProxyClient = Mockito.mock(MedicalClinicProxyClient.class);
+        this.doctorService = new DoctorService(medicalClinicProxyClient);
+    }
+
+    @Test
+    void getAvailableVisitsForDoctor_DataCorrect_ListVisitDtoReturned() {
+        VisitDto visitDto1 = new VisitDto(1L,
+                LocalDateTime.of(2024, 9, 22, 16, 30, 00),
+                LocalDateTime.of(2024, 9, 22, 17, 00, 00),
+                5L, 7L);
+        VisitDto visitDto2 = new VisitDto(2L,
+                LocalDateTime.of(2024, 10, 22, 16, 30, 00),
+                LocalDateTime.of(2024, 10, 22, 17, 00, 00),
+                5L, 11L);
+        List<VisitDto> visitDtos = List.of(visitDto1, visitDto2);
+
+        when(medicalClinicProxyClient.getAvailableVisitsForDoctor(5L)).thenReturn(visitDtos);
+
+        List<VisitDto> result = doctorService.getAvailableVisitsForDoctor(5L);
+
+        assertNotNull(result);
```

Comment:
nie wiem czy jest sens aż tylu asercji, skoro i tak sprawdzamy size listy i wartości zerowego indeksu, to moglibyśmy się ograniczyć do zrobienia listy tylko z 1 obiektem i sprawdzić tylko 0 indeks, co da nam jakieś 10 linijek mniej kodu przy każdym teście na napisanie dodatkowego buildera i wszystkich asercji

Reply:
racja, w sumie niepotrzebnie az tyle